src/components/settings/PersonalizedPushSettings.tsx

```diff
@@ -124,15 +144,9 @@ export const PersonalizedPushSettings = () => {
 
   const listItems = useMemo(() => {
     const parsedListItems = data?.categories?.map((category: Category) => {
-      const tagListArray = Array.isArray(category.tagList)
-        ? category.tagList
-        : typeof category.tagList === 'string'
-        ? category.tagList.split(',').map((tag) => tag.trim())
-        : [];
+      const tags = tagListArray(category.tagList);
 
-      const isExcluded = tagListArray.some(
-        (tag) => excludeCategoryIds?.[tag]?.[category.id] !== undefined
-      );
+      const isExcluded = tags.some((tag) => excludeCategoryIds?.[tag]?.[category.id] !== undefined);
```

Comment:
why `!== undefined`? what are the values here too look for with `some`?

Reply:
The matching values here are undefined or an empty object. If it is an empty object then the isExcluded value will be true, otherwise we will have a false value.